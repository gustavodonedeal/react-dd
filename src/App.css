.page-row {
  max-width: 1234px;
  /*margin: 0 auto;*/
  width: 100%;
  /*padding: 0 2.5%;*/
}
.header-content {
  color: rgba(0, 0, 0, .75);
}
.header-content a {
  color: inherit;
}
.menu-bar-container {
  margin: 0 auto;
  background-color: #fff;
  border-bottom: 5px solid #f0292a;
}
.menu-bar-content {
  height: 70px;
}
.header-logo{
  float: left;
  margin-right: 10px;
  display: block;
  margin-right: 12px;
  height: 32px;
  min-width: 156px;
  position: relative;
  top: 50%;
  -webkit-transform: translateY(-50%);
  transform: translateY(-50%);
}
.header-logo-image{
  display: inline-block;
  height: 100%;
  width: 156px;
  max-width: 156px;
}


.card-item {
  margin-bottom: 14px;
}
.card__link,
.card__link:hover {
  text-decoration: none;
}
.card{
  display: flex;
  position: relative;
  background: #fff;
  box-shadow: 0 1px 1px 0 rgba(0, 0, 0, .2);
  border-radius: 2px;
  overflow: hidden;
  text-align: left;
}
.card__media{
  min-width: 200px;
  -webkit-box-flex: 1;
  flex: 1 1 200px;
  max-width: 300px;
  position: relative;
}
.card__photo{
  background-color: #e0e0db;
  position: relative;
  display: block;
  width: 100%;
  height: 0;
  padding: 75% 0 0;
  overflow: hidden;
}
.card__photo img{
  position: absolute;
  display: block;
  max-width: 100%;
  max-height: 100%;
  left: 0;
  right: 0;
  top: 0;
  bottom: 0;
  margin: auto;
}
.card__media-count{
  position: absolute;
  left: 10px;
  bottom: 10px;
  padding: 8px;
  background: rgba(0, 0, 0, .4);
  border-radius: 4px;
  color: hsla(0, 0%, 100%, .6);
  font-size: 13px;
  font-weight: 400;
}
.card__media-count .icon-camera{
  display: inline-block;
  background-size: contain;
  height: 19px;
  width: 19px;
  margin: -4px 4px -4px 0;
}
.icon-camera{
  background-image: url(data:image/svg+xml,%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20xmlns%3Axlink%3D%22http%3A%2F%2Fwww.w3.org%2F1999%2Fxlink%22%20version%3D%221.1%22%20id%3D%22camera-Layer_1%22%20x%3D%220px%22%20y%3D%220px%22%20viewBox%3D%220%200%2013.4%2010.3%22%20style%3D%22enable-background%3Anew%200%200%2013.4%2010.3%3B%22%20xml%3Aspace%3D%22preserve%22%20width%3D%2232%22%20height%3D%2232%22%3E%3Cstyle%20type%3D%22text%2Fcss%22%3E%09.st0%7Bopacity%3A0.8%3B%7D%09.st1%7Bfill%3A%23FFFFFF%3B%7D%3C%2Fstyle%3E%3Cg%20id%3D%22camera-XsEodM.tif_1_%22%20class%3D%22st0%22%3E%09%3Cg%3E%09%09%3Cg%3E%09%09%09%3Cpath%20class%3D%22st1%22%20d%3D%22M3.7%2C1.8C3.8%2C0.9%2C4.3%2C0.3%2C5%2C0.1C5.2%2C0%2C5.4%2C0%2C5.5%2C0C6.3%2C0%2C7%2C0%2C7.7%2C0c1%2C0%2C1.9%2C0.6%2C1.9%2C1.8c0.1%2C0%2C0.1%2C0%2C0.2%2C0%20%20%20%20%20c0.6%2C0%2C1.2%2C0%2C1.8%2C0c1%2C0%2C1.8%2C0.8%2C1.8%2C1.8c0%2C1.7%2C0%2C3.3%2C0%2C5c0%2C1-0.8%2C1.8-1.8%2C1.8c-3.3%2C0-6.6%2C0-9.9%2C0c-1%2C0-1.7-0.8-1.7-1.8%20%20%20%20%20c0-1.7%2C0-3.4%2C0-5c0-1%2C0.7-1.8%2C1.7-1.8c0.6%2C0%2C1.2%2C0%2C1.8%2C0C3.6%2C1.8%2C3.7%2C1.8%2C3.7%2C1.8z%20M6.7%2C3C5%2C3%2C3.6%2C4.3%2C3.6%2C6%20%20%20%20%20c0%2C1.7%2C1.4%2C3.1%2C3.1%2C3.1c1.7%2C0%2C3.1-1.4%2C3.1-3.1C9.8%2C4.3%2C8.4%2C3%2C6.7%2C3z%22%2F%3E%09%09%09%3Cpath%20class%3D%22st1%22%20d%3D%22M6.7%2C7.8C5.7%2C7.8%2C5%2C7%2C5%2C6c0-1%2C0.8-1.7%2C1.7-1.7c1%2C0%2C1.7%2C0.8%2C1.7%2C1.7C8.4%2C7%2C7.7%2C7.8%2C6.7%2C7.8z%22%2F%3E%09%09%3C%2Fg%3E%09%3C%2Fg%3E%3C%2Fg%3E%3C%2Fsvg%3E);
  background-repeat: no-repeat;
}
.card__body{
  min-width: 280px;
  -webkit-box-flex: 2;
  flex: 2 1 570px;
  max-width: none;
  padding: 12px 10px 12px 20px;
  display: flex;
  -webkit-box-direction: normal;
  -webkit-box-orient: vertical;
  flex-direction: column;
  -webkit-box-pack: justify;
  justify-content: space-between;
}
.card__body-title{
  font-size: 18px;
  color: rgba(0, 0, 0, .75);
  margin-bottom: 0;
}
p{
  margin: .4em 0;
  line-height: 1.2;
}
.card__adinfo{
  display: none;
}
ol, ul, li{
  list-style: none;
  padding: 0;
  margin: 0;
}
.card__adinfo li{
  display: inline-block;
  color: #999;
  color: rgba(0, 0, 0, .4);
  font-weight: 400;
  font-size: 11px;
  text-transform: uppercase;
}
.card__body-keyinfo{
  margin: .3em 0 .2em;
  text-align: left;
}
.card__body-keyinfo li:first-of-type{
  margin-left: 0;
}
.card__body-keyinfo li{
  color: #999;
  color: rgba(0, 0, 0, .4);
  font-size: 13px;
  display: inline-block;
}
.card__body-keyinfo li:first-of-type{
  margin-left: 0;
}
.card__body-keyinfo li:first-of-type::before{
  content: '';
}
.card__body-keyinfo li::before{
  content: "\000B7";
  margin: 0 4px;
}
.card__price{
  display: inline-block;
  margin: 10px 0 0;
  color: #1b1b1b;
  font-size: 20px;
  font-weight: 700;
}
.card__actions{
  position: absolute;
  bottom: 12px;
  right: 10px;
  z-index: 50;
}
.card__actions .icon{
  display: block;
  width: 23px;
  height: 23px;
  background-size: contain;
}
.icon-save-heart{
  background-image: url(data:image/svg+xml,%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg…20%20%20C16.9-14.8%2C14.9-12.5%2C13.4-10.5z%22%2F%3E%3C%2Fg%3E%3C%2Fsvg%3E);
  background-repeat: no-repeat;
}
.card__actions .icon:before{
  content: '';
  display: block;
  width: 36px;
  height: 36px;
  position: absolute;
  bottom: -6px;
  right: -6px;
}


.seller-interaction-container{
  text-align: center;
  padding: 15px 3% 0;
  background-color: #fff;
  box-shadow: 0 1px 1px 0 rgba(0, 0, 0, .2);
}
.contact-seller-container{
  display: inline-block;
  vertical-align: middle;
  min-width: 45%;
  max-width: 94%;
  width: -webkit-calc((600px - 93%) * 600);
  width: calc((600px - 93%) * 600);
  margin: 0 3% 28px;
  left: 0;
  position: relative;
  background: #fff;
  text-align: left;
  margin-bottom: 16px;
  padding-bottom: 10px;
}
.contact-seller-header{
  text-align: center;
}
.contact-seller-header .contact-seller-image{
  display: block;
  height: 47px;
  width: 47px;
  background-size: 100%;
  margin: 12px auto;
}
.icon-user_tick{
  background-image: url(data:image/svg+xml,%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg…9%2C0L33.55%2C7A0.4%2C0.4%2C0%2C0%2C1%2C33.67%2C7.33Z%22%2F%3E%3C%2Fsvg%3E);
  background-repeat: no-repeat;
}
.contact-seller-title{
  font-size: 24px;
  line-height: 115%;
  margin: 0 0 4px;
  word-wrap: break-word;
}
.seller-type-tag.trader{
  background: #ff9700;
}
.seller-type-tag{
  display: inline-block;
  padding: 6px;
  margin-top: 7px;
  text-align: center;
  font-size: 13px;
  font-weight: 700;
  color: #fff;
  border-radius: 2px;
}
a:link, a:visited{
  color: #0af;
}
.see-all-ads-link{
  display: inline-block;
  padding: 9px;
  margin-bottom: 5px;
}
a{
  text-decoration: none;
}
.contact-seller-method{
  margin-top: 10px;
  min-height: 44px;
  height: auto;
}
.message-history{
  text-align: center;
  padding: 12px;
  margin-bottom: 1em;
  background: #fafafa;
  font-size: 15px;
  color: #999;
  color: rgba(0, 0, 0, .4);
  font-weight: 400;
  border-radius: 2px;
}
.message-history label{
  display: block;
  font-weight: 700;
  margin-bottom: .6ex;
}
  .btn, a.btn{
    height: 44px;
    line-height: 41px;
    border-radius: 4px;
    box-shadow: 0 3px 1px rgba(0, 0, 0, .05);
    font-size: 18px;
    padding: 0 3ex;
  }
.btn, a.btn{
  cursor: pointer;
  display: inline-block;
  vertical-align: middle;
  font-weight: 400;
  text-decoration: none;
  text-align: center;
  height: 36px;
  line-height: 34px;
  border-radius: 3px;
  box-shadow: 0 2px 1px rgba(0, 0, 0, .05);
  font-size: 15px;
  padding: 0 3ex;
  color: #fff;
  border: 1px solid;
  border-color: #0087cc;
  background: #00b3ff;
  background: -webkit-linear-gradient(90deg, #00a9ff, #00bfff);
  -ms-filter: "progid:DXImageTransform.Microsoft.gradient(startColorstr='#00A9FF', endColorstr='#00BFFF' ,GradientType=0)";
  background: -webkit-linear-gradient(bottom, #00a9ff, #00bfff);
  background: linear-gradient(0deg, #00a9ff, #00bfff);
}
  .input-btn, .input-btn:hover, a.input-btn{
    -ms-filter: "progid:DXImageTransform.Microsoft.gradient(startColorstr='#EDEDE9', endColorstr='#FFFFFF' ,GradientType=0)";
  }
  .input-btn, a.input-btn{
    border-radius: 4px;
    font-size: 17px;
    color: #3f3f3f;
    color: rgba(0, 0, 0, .75);
    font-weight: 400;
    font-weight: 700;
    border: 1px solid #bbb;
    border-bottom: 1px solid #aaa;
    background: #fafafa;
    background: -webkit-linear-gradient(90deg, #edede9, #f6f6f4 20.26%, #fdfdfc 50.18%, #fff);
    background: -webkit-linear-gradient(bottom, #edede9, #f6f6f4 20.26%, #fdfdfc 50.18%, #fff);
    background: linear-gradient(0deg, #edede9, #f6f6f4 20.26%, #fdfdfc 50.18%, #fff);
  }
.btn.disabled, .btn[disabled]:not([disabled=false]), .btn[disabled]:not([disabled=false]):active, .btn[disabled]:not([disabled=false]):hover, a.btn.disabled, a.btn[disabled]:not([disabled=false]), a.btn[disabled]:not([disabled=false]):active, a.btn[disabled]:not([disabled=false]):hover{
  color: #ededed;
  border: 1px solid #ccc;
  box-shadow: none;
  background: #ccc;
  top: 0;
  cursor: default;
}
.contact-seller-message-submit-btn{
  width: 100%;
}
.contact-seller-by-phone-btn{
  display: block;
  height: auto;
  min-height: 36px;
  line-height: 36px;
  font-size: 17px;
  word-break: break-all;
}
.contact-seller-by-phone .show-number{
  font-size: 15px;
  display: block;
  text-align: center;
  background: none;
  border: none;
  width: 100%;
  padding: 0;
  line-height: 34px;
  color: inherit;
}
.seller-details-container{
  border-top: 0;
  padding-top: 0;
}
.seller-details-container{
  display: inline-block;
  vertical-align: middle;
  min-width: 42%;
  max-width: 94%;
  width: -webkit-calc((600px - 93%) * 600);
  width: calc((600px - 93%) * 600);
  position: relative;
  margin: 0 3% 14px;
}
.seller-details-container:before{
  content: '';
  display: inline-block;
  position: absolute;
  left: -6%;
  height: 100%;
  max-width: 1px;
  width: -webkit-calc((380px - 100%) * 380);
  width: calc((380px - 100%) * 380);
  background-color: #dedede;
}
.seller-details-content{
  width: 94%;
  left: 17px;
}
.seller-details-content{
  display: table;
  padding: 0;
  position: relative;
}
.seller-detail{
  display: table-row;
  width: 100%;
}
.seller-detail-label{
  color: #999;
  color: rgba(0, 0, 0, .4);
  text-align: left;
  vertical-align: middle;
}
.seller-detail-label, .seller-detail-value{
  display: table-cell;
  padding: 14px 0;
}
.seller-detail-value{
  line-height: 1.4;
  text-align: right;
}
.seller-detail-label, .seller-detail-value{
  display: table-cell;
  padding: 14px 0;
}
.seller-detail-value address{
  font-style: normal;
}
.seller-verification-status{
  display: table-row;
  width: 100%;
  color: #999;
  color: rgba(0, 0, 0, .4);
}
.seller-verification-status .verification-credential.email{
  text-align: left;
}
.seller-verification-status .verification-credential{
  display: table-cell;
  padding: 21px 0;
}
.seller-verification-status .verification-credential .icon{
  display: inline-block;
  height: 1em;
  width: 1em;
  background-size: 100%;
  vertical-align: top;
  margin-right: 1ex;
}
.icon-tick_green, .icon-tick_green\:regular{
  background-image: url(data:image/svg+xml,%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20id%3D%22tick_green-Layer_1%22%20data-name%3D%22Layer%201%22%20viewBox%3D%220%200%2013%2010%22%20width%3D%2232%22%20height%3D%2232%22%3E%3Cdefs%3E%3Cstyle%3E.cls-1%7Bfill%3A%237ac943%3B%7D%3C%2Fstyle%3E%3C%2Fdefs%3E%3Ctitle%3Ejenny_%3C%2Ftitle%3E%3Cpath%20class%3D%22cls-1%22%20d%3D%22M11.62%2C3.06a0.67%2C0.67%2C0%2C0%2C1-.2.49L6.17%2C8.81l-1%2C1a0.7%2C0.7%2C0%2C0%2C1-1%2C0l-1-1L0.58%2C6.18a0.7%2C0.7%2C0%2C0%2C1%2C0-1l1-1A0.67%2C0.67%2C0%2C0%2C1%2C2.06%2C4a0.67%2C0.67%2C0%2C0%2C1%2C.49.2L4.69%2C6.35%2C9.45%2C1.58a0.7%2C0.7%2C0%2C0%2C1%2C1%2C0l1%2C1A0.67%2C0.67%2C0%2C0%2C1%2C11.62%2C3.06Z%22%2F%3E%3C%2Fsvg%3E);
  background-repeat: no-repeat;
}
.seller-verification-status .verification-credential.phone{
  text-align: right;
}
.seller-verification-status .verification-credential{
  display: table-cell;
  padding: 21px 0;
}
.seller-verification-status .verification-credential .icon{
  display: inline-block;
  height: 1em;
  width: 1em;
  background-size: 100%;
  vertical-align: top;
  margin-right: 1ex;
}
.desc{
  white-space: pre-wrap;
}
