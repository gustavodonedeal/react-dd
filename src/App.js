import React, { Component } from 'react';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import history from './history';
import Search from './components/Search';
import View from './components/View';
import Header from './components/Header';

class App extends Component  {

  constructor() {
    super();

    this.updateState = this.updateState.bind(this);

    this.state = {
      ads: [],
      loading: true,
      filter: {}
    }
  }

  updateState(newState){
    this.setState(newState);
  }

  searchComponent(props){
    return (<Search {...props} {...this.state} updateState={this.updateState} />)
  }

  render(){
    return(
      <Router history={history}>
        <div>
          <Header />
          <Route exact path='/' component={(props) => this.searchComponent(props)}/>
          <Route exact path='/ad/:adId' component={View}/>
          <Route exact path='/:section' component={(props) => this.searchComponent(props)}/>
        </div>
      </Router>
    )
  }
}

export default App;
