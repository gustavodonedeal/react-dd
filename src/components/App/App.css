
p{
  margin: .4em 0;
  line-height: 1.2;
}
.card__adinfo{
  display: none;
}
ol, ul, li{
  list-style: none;
  padding: 0;
  margin: 0;
}
.card__adinfo li{
  display: inline-block;
  color: #999;
  color: rgba(0, 0, 0, .4);
  font-weight: 400;
  font-size: 11px;
  text-transform: uppercase;
}
.card__body-keyinfo{
  margin: .3em 0 .2em;
  text-align: left;
}
.card__body-keyinfo li:first-of-type{
  margin-left: 0;
}
.card__body-keyinfo li{
  color: #999;
  color: rgba(0, 0, 0, .4);
  font-size: 13px;
  display: inline-block;
}
.card__body-keyinfo li:first-of-type{
  margin-left: 0;
}
.card__body-keyinfo li:first-of-type::before{
  content: '';
}
.card__body-keyinfo li::before{
  content: "\000B7";
  margin: 0 4px;
}
.card__price{
  display: inline-block;
  margin: 10px 0 0;
  color: #1b1b1b;
  font-size: 20px;
  font-weight: 700;
}
.card__actions{
  position: absolute;
  bottom: 12px;
  right: 10px;
  z-index: 50;
}
.card__actions .icon{
  display: block;
  width: 23px;
  height: 23px;
  background-size: contain;
}
.icon-save-heart{
  background-image: url(data:image/svg+xml,%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg…20%20%20C16.9-14.8%2C14.9-12.5%2C13.4-10.5z%22%2F%3E%3C%2Fg%3E%3C%2Fsvg%3E);
  background-repeat: no-repeat;
}
.card__actions .icon:before{
  content: '';
  display: block;
  width: 36px;
  height: 36px;
  position: absolute;
  bottom: -6px;
  right: -6px;
}


.seller-interaction-container{
  text-align: center;
  padding: 15px 3% 0;
  background-color: #fff;
  box-shadow: 0 1px 1px 0 rgba(0, 0, 0, .2);
}
.contact-seller-container{
  display: inline-block;
  vertical-align: middle;
  min-width: 45%;
  max-width: 94%;
  width: -webkit-calc((600px - 93%) * 600);
  width: calc((600px - 93%) * 600);
  margin: 0 3% 28px;
  left: 0;
  position: relative;
  background: #fff;
  text-align: left;
  margin-bottom: 16px;
  padding-bottom: 10px;
}
.contact-seller-header{
  text-align: center;
}
.contact-seller-header .contact-seller-image{
  display: block;
  height: 47px;
  width: 47px;
  background-size: 100%;
  margin: 12px auto;
}
.icon-user_tick{
  background-image: url(data:image/svg+xml,%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg…9%2C0L33.55%2C7A0.4%2C0.4%2C0%2C0%2C1%2C33.67%2C7.33Z%22%2F%3E%3C%2Fsvg%3E);
  background-repeat: no-repeat;
}
.contact-seller-title{
  font-size: 24px;
  line-height: 115%;
  margin: 0 0 4px;
  word-wrap: break-word;
}
.seller-type-tag.trader{
  background: #ff9700;
}
.seller-type-tag{
  display: inline-block;
  padding: 6px;
  margin-top: 7px;
  text-align: center;
  font-size: 13px;
  font-weight: 700;
  color: #fff;
  border-radius: 2px;
}
a:link, a:visited{
  color: #0af;
}
.see-all-ads-link{
  display: inline-block;
  padding: 9px;
  margin-bottom: 5px;
}
a{
  text-decoration: none;
}
.contact-seller-method{
  margin-top: 10px;
  min-height: 44px;
  height: auto;
}
.message-history{
  text-align: center;
  padding: 12px;
  margin-bottom: 1em;
  background: #fafafa;
  font-size: 15px;
  color: #999;
  color: rgba(0, 0, 0, .4);
  font-weight: 400;
  border-radius: 2px;
}
.message-history label{
  display: block;
  font-weight: 700;
  margin-bottom: .6ex;
}
  .btn, a.btn{
    height: 44px;
    line-height: 41px;
    border-radius: 4px;
    box-shadow: 0 3px 1px rgba(0, 0, 0, .05);
    font-size: 18px;
    padding: 0 3ex;
  }
.btn, a.btn{
  cursor: pointer;
  display: inline-block;
  vertical-align: middle;
  font-weight: 400;
  text-decoration: none;
  text-align: center;
  height: 36px;
  line-height: 34px;
  border-radius: 3px;
  box-shadow: 0 2px 1px rgba(0, 0, 0, .05);
  font-size: 15px;
  padding: 0 3ex;
  color: #fff;
  border: 1px solid;
  border-color: #0087cc;
  background: #00b3ff;
  background: -webkit-linear-gradient(90deg, #00a9ff, #00bfff);
  -ms-filter: "progid:DXImageTransform.Microsoft.gradient(startColorstr='#00A9FF', endColorstr='#00BFFF' ,GradientType=0)";
  background: -webkit-linear-gradient(bottom, #00a9ff, #00bfff);
  background: linear-gradient(0deg, #00a9ff, #00bfff);
}
  .input-btn, .input-btn:hover, a.input-btn{
    -ms-filter: "progid:DXImageTransform.Microsoft.gradient(startColorstr='#EDEDE9', endColorstr='#FFFFFF' ,GradientType=0)";
  }
  .input-btn, a.input-btn{
    border-radius: 4px;
    font-size: 17px;
    color: #3f3f3f;
    color: rgba(0, 0, 0, .75);
    font-weight: 400;
    font-weight: 700;
    border: 1px solid #bbb;
    border-bottom: 1px solid #aaa;
    background: #fafafa;
    background: -webkit-linear-gradient(90deg, #edede9, #f6f6f4 20.26%, #fdfdfc 50.18%, #fff);
    background: -webkit-linear-gradient(bottom, #edede9, #f6f6f4 20.26%, #fdfdfc 50.18%, #fff);
    background: linear-gradient(0deg, #edede9, #f6f6f4 20.26%, #fdfdfc 50.18%, #fff);
  }
.btn.disabled, .btn[disabled]:not([disabled=false]), .btn[disabled]:not([disabled=false]):active, .btn[disabled]:not([disabled=false]):hover, a.btn.disabled, a.btn[disabled]:not([disabled=false]), a.btn[disabled]:not([disabled=false]):active, a.btn[disabled]:not([disabled=false]):hover{
  color: #ededed;
  border: 1px solid #ccc;
  box-shadow: none;
  background: #ccc;
  top: 0;
  cursor: default;
}
.contact-seller-message-submit-btn{
  width: 100%;
}
.contact-seller-by-phone-btn{
  display: block;
  height: auto;
  min-height: 36px;
  line-height: 36px;
  font-size: 17px;
  word-break: break-all;
}
.contact-seller-by-phone .show-number{
  font-size: 15px;
  display: block;
  text-align: center;
  background: none;
  border: none;
  width: 100%;
  padding: 0;
  line-height: 34px;
  color: inherit;
}
.seller-details-container{
  border-top: 0;
  padding-top: 0;
}
.seller-details-container{
  display: inline-block;
  vertical-align: middle;
  min-width: 42%;
  max-width: 94%;
  width: -webkit-calc((600px - 93%) * 600);
  width: calc((600px - 93%) * 600);
  position: relative;
  margin: 0 3% 14px;
}
.seller-details-container:before{
  content: '';
  display: inline-block;
  position: absolute;
  left: -6%;
  height: 100%;
  max-width: 1px;
  width: -webkit-calc((380px - 100%) * 380);
  width: calc((380px - 100%) * 380);
  background-color: #dedede;
}
.seller-details-content{
  width: 94%;
  left: 17px;
}
.seller-details-content{
  display: table;
  padding: 0;
  position: relative;
}
.seller-detail{
  display: table-row;
  width: 100%;
}
.seller-detail-label{
  color: #999;
  color: rgba(0, 0, 0, .4);
  text-align: left;
  vertical-align: middle;
}
.seller-detail-label, .seller-detail-value{
  display: table-cell;
  padding: 14px 0;
}
.seller-detail-value{
  line-height: 1.4;
  text-align: right;
}
.seller-detail-label, .seller-detail-value{
  display: table-cell;
  padding: 14px 0;
}
.seller-detail-value address{
  font-style: normal;
}
.seller-verification-status{
  display: table-row;
  width: 100%;
  color: #999;
  color: rgba(0, 0, 0, .4);
}
.seller-verification-status .verification-credential.email{
  text-align: left;
}
.seller-verification-status .verification-credential{
  display: table-cell;
  padding: 21px 0;
}
.seller-verification-status .verification-credential .icon{
  display: inline-block;
  height: 1em;
  width: 1em;
  background-size: 100%;
  vertical-align: top;
  margin-right: 1ex;
}
.icon-tick_green, .icon-tick_green\:regular{
  background-image: url(data:image/svg+xml,%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20id%3D%22tick_green-Layer_1%22%20data-name%3D%22Layer%201%22%20viewBox%3D%220%200%2013%2010%22%20width%3D%2232%22%20height%3D%2232%22%3E%3Cdefs%3E%3Cstyle%3E.cls-1%7Bfill%3A%237ac943%3B%7D%3C%2Fstyle%3E%3C%2Fdefs%3E%3Ctitle%3Ejenny_%3C%2Ftitle%3E%3Cpath%20class%3D%22cls-1%22%20d%3D%22M11.62%2C3.06a0.67%2C0.67%2C0%2C0%2C1-.2.49L6.17%2C8.81l-1%2C1a0.7%2C0.7%2C0%2C0%2C1-1%2C0l-1-1L0.58%2C6.18a0.7%2C0.7%2C0%2C0%2C1%2C0-1l1-1A0.67%2C0.67%2C0%2C0%2C1%2C2.06%2C4a0.67%2C0.67%2C0%2C0%2C1%2C.49.2L4.69%2C6.35%2C9.45%2C1.58a0.7%2C0.7%2C0%2C0%2C1%2C1%2C0l1%2C1A0.67%2C0.67%2C0%2C0%2C1%2C11.62%2C3.06Z%22%2F%3E%3C%2Fsvg%3E);
  background-repeat: no-repeat;
}
.seller-verification-status .verification-credential.phone{
  text-align: right;
}
.seller-verification-status .verification-credential{
  display: table-cell;
  padding: 21px 0;
}
.seller-verification-status .verification-credential .icon{
  display: inline-block;
  height: 1em;
  width: 1em;
  background-size: 100%;
  vertical-align: top;
  margin-right: 1ex;
}
.desc{
  white-space: pre-wrap;
}
