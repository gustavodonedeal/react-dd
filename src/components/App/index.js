import React, { Component } from 'react';
import { Link, Route } from 'react-router-dom';
import universal from 'react-universal-component';

const UniversalRoute = universal(({ component }) => import(`../${component}`));
const TestPage = props => (<UniversalRoute component={'TestPage'} {...props} />);
const AdvertPage = props => (<UniversalRoute component={'Advert'} {...props} />);

const App = () => (
    <div>
        <nav>
            <Link to='/test'>Hello</Link>
        </nav>
        <Route path='/test' component={TestPage} />
        <Route path='/ad/:adId' component={AdvertPage} />
    </div>
);

export default class extends Component {
    render() {
        return (
            <App />
        );
    }
}