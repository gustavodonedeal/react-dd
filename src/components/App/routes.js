import React from "react";
import universal from "react-universal-component";
import { fetchAdvert } from "../Advert/reducer";

const UniversalRoute = universal(({ component }) => import(`../${component}`));
const AdvertPage = props => <UniversalRoute component={"Advert"} {...props} />;

const routes = [
  {
    path: "/ad/:adId",
    component: AdvertPage,
    loadData: ({ params }) => fetchAdvert(params.adId)
  }
];

export default routes;
