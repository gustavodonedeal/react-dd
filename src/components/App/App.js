import React, { Component } from 'react';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import { createBrowserHistory } from 'history'
import Search from '../Search';
import View from '../View';
import Header from '../Header';
import './App.css';

class App extends Component  {

  constructor() {
    super();

    this.state = {
      ads: [],
      loading: true,
      filter: {}
    }
  }

  updateState = newState => this.setState(newState);

  searchComponent(props){
    return (<Search {...props} {...this.state} updateState={this.updateState} />)
  }

  render(){
    return(
      <Router history={createBrowserHistory()}>
        <div>
          <Header />
          <Route exact path='/' component={(props) => this.searchComponent(props)} />
          <Route exact path='/ad/:adId' component={View} />
          <Route exact path='/:section' component={(props) => this.searchComponent(props)}/>
        </div>
      </Router>
    )
  }
}

export default App;
