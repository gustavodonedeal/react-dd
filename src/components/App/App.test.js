import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import renderer from 'react-test-renderer';

it('renders inital loading state correctly', () => {
  const tree = renderer.create(
    <App />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
