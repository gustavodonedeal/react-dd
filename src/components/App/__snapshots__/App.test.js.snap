// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders inital loading state correctly 1`] = `
<div>
  <header
    className="header-container"
  >
    <div
      className="header-content"
    >
      <div
        className="menu-bar-container"
      >
        <div
          className="menu-bar-content page-row"
        >
          <a
            className="header-logo"
            href="/"
            onClick={[Function]}
          >
            <img
              alt="DoneDeal"
              className="header-logo-image"
              src="donedeal-logo.svg"
            />
          </a>
        </div>
      </div>
    </div>
  </header>
  <p>
    Loading...
  </p>
</div>
`;
